src/services/users.ts

```diff
@@ -77,25 +77,57 @@ export function updateUser(dbPool: PostgresJsDatabase<typeof db>) {
     }
 
     // update user
-    const user = await dbPool
-      .update(db.users)
-      .set({
-        email: body.data.email,
-        username: body.data.username,
-        name: body.data.name,
-        updatedAt: new Date(),
-      })
-      .where(eq(db.users.id, userId))
-      .returning();
-
-    const updatedGroups = await overwriteUsersToGroups(dbPool, userId, body.data.groupIds);
-
-    const updatedUserAttributes = await upsertUserAttributes(
-      dbPool,
-      userId,
-      body.data.userAttributes,
-    );
-
-    return res.json({ data: { user, updatedGroups, updatedUserAttributes } });
+    try {
+      // check if username or email already exists and is not the current user
+      if (body.data.email || body.data.username) {
```

Comment:
Does it take into account that the username can be `null`? For example. Iogin using Zupass but do not fill out account registration and log out. We have a few cases of those in production. The result is that username is `null` and email = email.

Reply:
it will still pass fine, added a check so frontend will not send empty strings but will convert it to null